OpenRA.Mods.Common/Traits/World/PathFinder.cs

```diff
@@ -93,12 +96,36 @@ public List<CPos> FindUnitPath(CPos source, CPos target, Actor self, Actor ignor
 			return pb;
 		}
 
-		public List<CPos> FindUnitPathToRange(CPos source, SubCell srcSub, WPos target, WDist range, Actor self, BlockedByActor check)
+		public List<CPos> FindUnitPath(CPos source, CPos target, Actor self, Actor ignoreActor, BlockedByActor check)
 		{
-			if (!cached)
+			using (new PerfSample("Pathfinder"))
+			{
+				var key = new PathCacheKey(self.ActorID, source, target);
+
+				// Only cache path when transient actors are ignored, otherwise there is no guarantee that the path
+				// is still valid at the next check.
+				if (check == BlockedByActor.None)
```

Comment:
Question that's been on my mind for a while. The cache was a minor but measurable benefit when originally added. It however didn't include this check at the time. Quite rightly we can't reuse paths when we need to check for blocking actors - so this bugfix is important.

How often do we even do such a path search though? Most of the searches are going to want to consider blocked actors on the map.

Is this cache even getting used?

Reply:
I did not check this. I would not know.

Reply:
Air perhaps?

Reply:
I'd be interested to see if we can get some statistics on it - if it's not used, or used so rarely the overhead from cache management outweighs the saving of not doing a path search 1 time in 100 (or whatever it is) then we could just bin the whole thing off rather than worrying about optimizing it.

Reply:
Did some checking and it seems the cache has a poor hit rate, so I filed #19815